Darwin: can't install mu

Looks like `mu` can't be installed because a temporary name can't be created:

```
Cannot create temporary name for prefix: Bad file descriptor, mu4e-actions.elc
```

I'll continue troubleshooting, but wanted to create an issue in case anyone else comes across this issue. I'm running the latest version of El Capitan `10.11.2 (15C50)`. To reproduce on my Mac I just have to run `nix-env -i mu`.

```
$ uname -a
Darwin mbp.local 15.2.0 Darwin Kernel Version 15.2.0: Fri Nov 13 19:56:56 PST 2015; root:xnu-3248.20.55~2/RELEASE_X86_64 x86_64

$ nix-env --version
nix-env (Nix) 1.10

$ git describe # in nixpkgs repo
15.09-beta-5857-gecc0c30
```

Full output can be found in this gist: https://gist.github.com/jcf/ce6a50b92dcdebac57ad

Same thing with NixOS/nix@32ea8a9:

```
make[3]: Entering directory '/private/var/folders/0_/ldrvd84j5b712wbnfrg7z_700000gn/T/nix-build-mu-0.9.13.drv-0/mu-0.9.13/mu4e'
  MAKEINFO mu4e.info
mu4e.texi:55: warning: undefined flag: mu-version.
cd .. && /nix/store/1qh7l7vb89hfgmai1c0fjh2r50fgyjrx-bash-4.3-p42/bin/bash ./config.status mu4e/mu4e-meta.el
config.status: creating mu4e/mu4e-meta.el
if test "emacs" != "no"; then \
  am__dir=. am__subdir_includes=''; \
  case mu4e-actions.elc in */*) \
    am__dir=`echo 'mu4e-actions.elc' | sed 's,/[^/]*$,,'`; \
    am__subdir_includes="-L $am__dir -L ./$am__dir"; \
  esac; \
  test -d "$am__dir" || /nix/store/cgyn13df4j2s8aksv8miaz0yimvldmnk-coreutils-8.24/bin/mkdir -p "$am__dir" || exit 1; \
  emacs --batch \
      \
    $am__subdir_includes -L . -L . \
    --eval "(defun byte-compile-dest-file (f) \"mu4e-actions.elc\")" \
    --eval "(unless (byte-compile-file \"mu4e-actions.el\") (kill-emacs 1))"; \
else :; fi
Cannot create temporary name for prefix: Bad file descriptor, mu4e-actions.elc
Makefile:631: recipe for target 'mu4e-actions.elc' failed
make[3]: *** [mu4e-actions.elc] Error 255
make[3]: Leaving directory '/private/var/folders/0_/ldrvd84j5b712wbnfrg7z_700000gn/T/nix-build-mu-0.9.13.drv-0/mu-0.9.13/mu4e'
Makefile:704: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory '/private/var/folders/0_/ldrvd84j5b712wbnfrg7z_700000gn/T/nix-build-mu-0.9.13.drv-0/mu-0.9.13/mu4e'
Makefile:494: recipe for target 'all-recursive' failed
```

Calling it a night. Hopefully I can get `mu` installed via Nix tomorrow!

--- a/pkgs/applications/editors/emacs-24/at-fdcwd.patch
+++ b/pkgs/applications/editors/emacs-24/at-fdcwd.patch
@@ -1,12 +1,14 @@
 diff --git a/lib/careadlinkat.h b/lib/careadlinkat.h
-index 5cdb813..7a272e8 100644
+index 84ede3e..8e8f42e 100644
 --- a/lib/careadlinkat.h
 +++ b/lib/careadlinkat.h
-@@ -23,6 +23,8 @@
+@@ -23,6 +23,10 @@
  #include <fcntl.h>
  #include <unistd.h>
  
++#ifndef AT_FDCWD
 +#define AT_FDCWD -2
++#endif
 +
  struct allocator;
  

--- a/pkgs/applications/editors/emacs-25/at-fdcwd.patch
+++ b/pkgs/applications/editors/emacs-25/at-fdcwd.patch
@@ -1,12 +1,14 @@
 diff --git a/lib/careadlinkat.h b/lib/careadlinkat.h
-index 5cdb813..7a272e8 100644
+index 84ede3e..8e8f42e 100644
 --- a/lib/careadlinkat.h
 +++ b/lib/careadlinkat.h
-@@ -23,6 +23,8 @@
+@@ -23,6 +23,10 @@
  #include <fcntl.h>
  #include <unistd.h>
  
++#ifndef AT_FDCWD
 +#define AT_FDCWD -2
++#endif
 +
  struct allocator;
  

